ArnoldOptions log verbosity

johnhaddon opened the pull request:
This adds verbosity controls to the ArnoldOptions node. There's a slight change of behaviour in that now by default all logging to the console is turned off - this is the default at the Arnold library level. @andrewkaufman, I believe that's what you wanted for the ShaderView, but I don't know if it'll be non-ideal for interactive renders and batch renders?

andrewkaufman commented:
For sure that sounds good for the ShaderView. We can decide whats a good default for batch and ipr and set those up on the IE wrapper nodes.

andrewkaufman commented:
Regarding 690c1b8, I haven't built locally to see what this looks like, but aside from just a font change, could they be slightly more indented? It feels like its a 1 pixel offset right now, which is just barely enough to tell... to be honest I'm not even sure if it is indented more, I might just be seeing things.

johnhaddon commented:
I did try adding some indentation actually - it was easy enough to indent the text, and I presume I could the icon too. I was just a bit worried about the existence of use cases where it wasn't just a nesting inside a column, where it might not be appropriate to indent. I suspect Jabuka might us collapsibles in a bunch of ways Gaffer doesn't, for instance.

johnhaddon commented:
Oh wait, are you asking if they _are_ indented, or if I could add some indentation? I haven't done anything to add indentation, but I think you might be right, there might be a pixel creeping in from some unasked-for border.

andrewkaufman commented:
Yeah, I think I noticed a tiny tiny indentation and I wanted a bit more.

johnhaddon commented:
I'd be happy to add some more - I think it would work well everywhere in Gaffer. It would be good to get @ivanimanishi's thoughts though, since I don't know in what ways collapsibles are nested in Jabuka etc.

andrewkaufman commented:
@ivanimanishi was jumping for joy when I mentioned indenting them more. Its possible there is some use-case we're overlooking, but immediate reactions all seem positive.

johnhaddon commented:
OK, I've removed the teeny indent, replaced it with a proper indent on the collapsible label, and made the default console output include errors and warnings, but otherwise be silent (having played with it a bit, I found the silencing of errors to be very unhelpful).

andrewkaufman commented:
Great, thanks!
